server/routes/record-attendance/handlers/editAttendance.ts

```diff
@@ -0,0 +1,130 @@
+import { Request, Response } from 'express'
+import { areIntervalsOverlapping, parse } from 'date-fns'
+import { Expose } from 'class-transformer'
+import { IsIn } from 'class-validator'
+import ActivitiesService from '../../../services/activitiesService'
+import PrisonService from '../../../services/prisonService'
+import { toDate } from '../../../utils/utils'
+import { ScheduledActivity, ScheduledEvent } from '../../../@types/activitiesAPI/types'
+
+enum EditAttendanceOptions {
+  YES = 'yes',
+  NO = 'no',
+  REMOVE = 'remove',
+}
+
+export class EditAttendance {
+  @Expose()
+  @IsIn(Object.values(EditAttendanceOptions), { message: 'Select an attendance option' })
+  attendanceOption: string
+}
+
+export default class EditAttendanceRoutes {
+  constructor(private readonly activitiesService: ActivitiesService, private readonly prisonService: PrisonService) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const { id } = req.params
+    const { attendanceId } = req.params
+
+    const instance = await this.activitiesService.getScheduledActivity(+id, user).then(i => ({
+      ...i,
+    }))
+
+    const attendance = await this.activitiesService.getAttendanceDetails(+attendanceId, user)
+
+    const prisonerNumbers = [attendance.prisonerNumber]
+
+    const attendees = await this.prisonService.searchInmatesByPrisonerNumbers(prisonerNumbers, user).then(inmates =>
+      inmates.map(i => ({
+        name: `${this.capitalize(i.firstName)} ${this.capitalize(i.lastName)}`,
+      })),
+    )
+
+    const attendee = attendees[0]
+
+    res.render('pages/record-attendance/edit-attendance', { instance, attendance, attendee })
+  }
+
+  POST = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const { id } = req.params
+    const { attendanceId } = req.params
+    if (req.body.attendanceOption === 'yes') {
+      const attendances = [
+        {
+          id: +attendanceId,
+          status: 'COMPLETED',
+          attendanceReason: 'ATTENDED',
+          issuePayment: true,
+        },
+      ]
+      await this.activitiesService.updateAttendances(attendances, user)
+      return res.redirect(`/attendance/activities/${id}/attendance-details/${attendanceId}`)
+    }
+
+    if (req.body.attendanceOption === 'no') {
+      const attendance = await this.activitiesService.getAttendanceDetails(+attendanceId, user)
+
+      const instance = await this.activitiesService.getScheduledActivity(+id, user)
```

Comment:
Slightly more performant:

```suggestion
      const [attendance, instance] = await Promise.all([
          this.activitiesService.getAttendanceDetails(+attendanceId, user),
          this.activitiesService.getScheduledActivity(+id, user),
      ])
```